docs/howto/custom-template-tags.txt

```diff
@@ -498,6 +498,128 @@ you see fit:
+  so we receive a plain string.
+* If an argument was a template variable, our function is passed the current
+  value of the variable, not the variable itself.
+* Variables passed to the tag are not included in the rendering context of the
+  content, as would be using the ``{% with %}`` tag.
+
+Like ``simple_tag``, ``simple_block_tag`` passes its output through
+:func:`~django.utils.html.conditional_escape` if the template context is in
+autoescape mode, to ensure correct HTML and protect you from XSS
+vulnerabilities.
+
+If additional escaping is not desired, you will need to use
+:func:`~django.utils.safestring.mark_safe` if you are absolutely sure that your
+code does not contain XSS vulnerabilities. For building small HTML snippets,
+use of :func:`~django.utils.html.format_html` instead of ``mark_safe()`` is
+strongly recommended.
+
+Because the ``content`` argument has already been rendered by Django, it is
+already escaped. Therefore, it may be necessary to use
+:func:`~django.utils.safestring.mark_safe` to ensure it's not double-escaped.
+
+If your template tag needs to access the current context, you can use the
+``takes_context`` argument when registering your tag::
+
+    @register.simple_block_tag(takes_context=True)
+    def chart(context, content):
+        return render_chart(source=content, dark_mode=context["dark_mode"])
+
+Note that the first argument *must* be called ``context``, and ``content``
+moves to the second argument.
+
+For more information on how the ``takes_context`` option works, see the section
+on :ref:`inclusion tags<howto-custom-template-tags-inclusion-tags>`.
+
+If you need to rename your tag, you can provide a custom name for it::
+
+    register.simple_block_tag(lambda content: content.upper(), name="uppercase")
+
+
+    @register.simple_block_tag(name="lowercase")
+    def some_function(content):
+        return content.lower()
+
+``simple_block_tag`` functions may accept any number of positional or keyword
+arguments. For example::
+
+    @register.simple_block_tag
+    def mytag(content, a, b, *args, **kwargs):
+        profile = kwargs["profile"]
+        ...
+        return ...
+
+Then in the template any number of arguments, separated by spaces, may be
+passed to the template tag. Like in Python, the values for keyword arguments
+are set using the equal sign ("``=``") and must be provided after the
+positional arguments. For example:
+
+.. code-block:: html+django
+
+    {% mytag 123 "abcd" book.title profile=user.profile %}
+        This is some extra content.
+    {% endmytag%}
+
+It's possible to store the tag results in a template variable rather than
+directly outputting it. This is done by using the ``as`` argument followed by
+the variable name. Doing so enables you to output the content yourself where
+you see fit:
+
+.. code-block:: html+django
+
+    {% chart as my_chart %}
+        A -> {B C}
+    {% endchart %}
+    <p>The chart is {{ my_chart|length }} characters.</p>
+
```

Comment:
I think that there are too many bits that are duplicated from the "simple tags" explanation. I think those should be framed under a summary, helping reducing the duplication. How about something like:

```suggestion
* Variables passed to the tag are not included in the rendering context of the
  content, as would be when using the ``{% with %}`` tag.

Just like :ref:`simple_tag<howto-custom-template-tags-simple-tags>`,
``simple_block_tag``:

* Validates the quantity and quality of the arguments.
* Quotes are stripped from arguments if necessary.
* Output is escaped accordingly.
* Context can be accessed by passing ``takes_context=True`` at registration
  time. Note that in this case, the first argument *must* be called
  ``context``, and ``content`` must follows.
* The tag can be renamed using passing the ``name`` argument when registering.
* ``simple_block_tag`` functions may accept any number of positional or keyword
  arguments.
* The tag result can be stored in a template variable using the ``as`` variant.

.. admonition::  Content Escaping

    ``simple_block_tag`` behaves similarly to ``simple_tag`` regarding
    auto-escaping. For details on escaping and safety, refer to ``simple_tag``.

    Because the ``content`` argument has already been rendered by Django, it is
    already escaped. Therefore, it may be necessary to use
    :func:`~django.utils.safestring.mark_safe` to ensure it's not
    double-escaped.
```